rails_app/app/adapters/graphite.rb

```diff
@@ -0,0 +1,60 @@
+require 'net/http'
+require 'json'
+require 'graphite-api'
+
+class GraphiteAdapter
+    def initialize
+        @graphite_host = Settings.graphite.host
+        @default_period = Settings.graphite.default_get_metric_time
+        @api = GraphiteAPI.new(
+            graphite: "#{@graphite_host}:#{Settings.graphite.port}",
+            prefix: Settings.graphite.metric_prefix
+        )
+    end
+
+    def send_raw(metric, value = nil, time = nil)
+        if !metric.is_a?(Hash)
+            metric = { metric => value }
+        end
+
+        if time
```

Comment:
```time.present? ? @api.metrics(metric, Time.at(time)) : @api.metrics(metric)```

Reply:
the hell is this thing?
> An object is present if it’s not blank.
`. nil?` seems like a better choice

Reply:
could be